front/src/applications/operationalStudies/hooks/useInvalidTrainOps.ts

```diff
@@ -0,0 +1,64 @@
+import { useMemo } from 'react';
+
+import type {
+  PathPropertiesFormatted,
+  SimulationResults,
+} from 'applications/operationalStudies/types';
+import {
+  osrdEditoastApi,
+  type OperationalPointReference,
+  type OperationalPointPart,
+} from 'common/api/osrdEditoastApi';
+
+import { getStationFromOps, isOperationalPointReference } from '../utils';
+
+const useInvalidTrainOps = (
+  infraId: number,
+  simulationResults: SimulationResults | undefined
+): PathPropertiesFormatted['operationalPoints'] => {
+  const operationalPointReferences: OperationalPointReference[] = useMemo(() => {
+    if (!simulationResults || simulationResults.isValid) return [];
+    return simulationResults.train.path
+      .filter((pathItem) => isOperationalPointReference(pathItem))
+      .map(
+        ({ id: _id, deleted: _deleted, ...cleanOperationalPointReference }) =>
+          cleanOperationalPointReference
+      );
+  }, [simulationResults]);
+
+  const { data: operationalPoints } =
+    osrdEditoastApi.endpoints.postInfraByInfraIdMatchOperationalPoints.useQuery(
+      {
+        infraId,
+        body: {
+          operational_point_references: operationalPointReferences,
+        },
+      },
+      { skip: !infraId || operationalPointReferences.length === 0 }
```

Comment:
We can remove `infraId` as it's always defined
```suggestion
      { skip: operationalPointReferences.length === 0 }
```

Reply:
Yes, you're right, thanks!